crates/burn-import/src/onnx/to_burn.rs

```diff
@@ -433,6 +422,39 @@ impl OnnxGraph {
         RandomNormalNode::new(output_type, mean, scale)
     }
 
+    pub(crate) fn constant_of_shape_conversion(node: Node) -> ConstantOfShapeNode {
+        // Additional types needed for ConstantOfShape:
+        use crate::burn::node::constant_of_shape::ConstantValue;
+
+        let input = node
+            .inputs
+            .first()
+            .expect("ConstantOfShape requires an input tensor");
+        let output = node.outputs.first().unwrap();
+
+        let value = node
+            .attrs
+            .get("value")
+            .and_then(|val| val.clone().into_tensor().data)
+            .map(|val_data| match val_data {
+                // TODO: Handle Float16
+                Data::Float32(val) => val.into(),
+                Data::Float32s(vals) => ConstantValue::from_vec(vals),
+                Data::Float64(val) => val.into(),
+                Data::Float64s(vals) => ConstantValue::from_vec(vals),
+                Data::Int32(val) => val.into(),
+                Data::Int32s(vals) => ConstantValue::from_vec(vals),
+                Data::Int64(val) => val.into(),
+                Data::Int64s(vals) => ConstantValue::from_vec(vals),
+                Data::Bool(val) => val.into(),
+                Data::Bools(vals) => ConstantValue::from_vec(vals),
```

Comment:
We only need to handle scalar types, not vectors. The value is a single element, so no need for `Float32s`, `Float64s`, etc.

Reply:
Fixed

Reply:
Actually the data comes as 1d tensor (arrays) (see https://onnx.ai/onnx/operators/onnx__ConstantOfShape.html) so we need to handle arrays not scalars. I removed scalars.